src/tools/ceph_objectstore_tool.cc

```diff
@@ -2259,6 +2260,60 @@ int mydump_journal(Formatter *f, string journalpath, bool m_journal_dio)
   return r;
 }
 
+int apply_layout_settings(ObjectStore *os, const OSDSuperblock &superblock,
+			  const string &pool_name, bool dry_run)
+{
+  int r = 0;
+
+  OSDMap curmap;
+  bufferlist bl;
+  r = get_osdmap(os, superblock.current_epoch, curmap, bl);
+  if (r) {
+    cerr << "Can't find local OSDMap: " << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  int64_t poolid = curmap.lookup_pg_pool_name(pool_name);
+  if (poolid < 0) {
+    cerr << "Couldn't find pool " << pool_name << ": " << cpp_strerror(poolid)
+	 << std::endl;
+    return poolid;
+  }
+
+  vector<coll_t> collections, in_pool;
+  r = os->list_collections(collections);
+  if (r < 0) {
+    cerr << "Error listing collections: " << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  for (vector<coll_t>::const_iterator it = collections.begin();
+       it != collections.end(); ++it) {
+    spg_t pgid;
+    if (it->is_pg(&pgid) && pgid.pool() == (uint64_t)poolid) {
+      in_pool.push_back(*it);
```

Comment:
Is there any use to only splitting a particular pgid?  Because if --pgid option is specified instead of pool it could be passed to the code below as coll_t coll(pgid).